miden-rpc/src/server/api.rs

```diff
@@ -0,0 +1,122 @@
+            "DB client initialized",
+        );
+        Ok(Self {
+            client: Arc::new(client),
+        })
+    }
+}
+
+#[tonic::async_trait]
+impl rpc::api_server::Api for RpcApi {
+    type CheckNullifiersStream = Streaming<rpc::ResponseBytes>;
+    type ListNotesByTagStream = Streaming<rpc::ResponseBytes>;
+
+    async fn get_block_header(
+        &self,
+        request: Request<rpc::GetBlockHeaderRequest>,
+    ) -> Result<Response<rpc::ResponseBytes>, Status>
+    {
+        let mut client = self.client.as_ref().clone();
+        let inner = request.into_inner();
+        let request = store::GetBlockHeaderRequest{block_spec: inner.block_spec};
+        let response = client.get_block_header(request).await?;
+        let data = response.into_inner().encode_to_vec();
+        Ok(Response::new(rpc::ResponseBytes{data}))
+    }
+
+    async fn get_note(
+        &self,
+        request: Request<rpc::GetNoteRequest>,
+    ) -> Result<Response<rpc::ResponseBytes>, Status>
+    {
+        let mut client = self.client.as_ref().clone();
+        let inner = request.into_inner();
+        let request = store::GetNoteRequest{note_spec: inner.note_spec};
+        let response = client.get_note(request).await?;
+        let data = response.into_inner().encode_to_vec();
+        Ok(Response::new(rpc::ResponseBytes{data}))
+    }
+
+    async fn check_nullifiers(
+        &self,
+        _request: Request<Streaming<digest::Digest>>,
+    ) -> Result<Response<Self::CheckNullifiersStream>, Status>
+    {
+        todo!()
+    }
+
+    async fn get_transaction_data(
+        &self,
+        request: Request<rpc::GetTransactionDataRequest>,
+    ) -> Result<Response<rpc::ResponseBytes>, Status>
+    {
+        let mut client = self.client.as_ref().clone();
+        let inner = request.into_inner();
+        let request = store::GetTransactionDataRequest{
+            account_id: inner.account_id,
+            block_spec: inner.block_spec,
+            notes: inner.notes,
+        };
+        let response = client.get_transaction_data(request).await?;
+        let data = response.into_inner().encode_to_vec();
+        Ok(Response::new(rpc::ResponseBytes{data}))
+    }
+
+    async fn list_notes_by_tag(
+        &self,
+        _request: Request<rpc::ListNotesByTagRequest>,
+    ) -> Result<Response<Self::ListNotesByTagStream>, Status>
+    {
+        todo!()
+    }
+
+    async fn submit_transaction(
+        &self,
+        request: Request<rpc::SubmitTransactionRequest>,
+    ) -> Result<Response<rpc::ResponseBytes>, Status>
+    {
+        let mut client = self.client.as_ref().clone();
+        let inner = request.into_inner();
+        let request = store::SubmitTransactionRequest{
+            transaction: inner.transaction,
+        };
+        let response = client.submit_transaction(request).await?;
+        let data = response.into_inner().encode_to_vec();
+        Ok(Response::new(rpc::ResponseBytes{data}))
+    }
```

Comment:
I would leave the implementation as `todo!()` for now as `submit_transaction()` request not go to the database, but to the mempool component.

Reply:
removed